core/primitives/src/test_utils.rs

```diff
@@ -28,42 +29,47 @@ where
     f();
 }
 
+fn setup_subscriber_from_filter(mut env_filter: EnvFilter) {
+    if let Ok(rust_log) = env::var("RUST_LOG") {
+        for directive in rust_log.split(',').filter_map(|s| match s.parse() {
+            Ok(directive) => Some(directive),
+            Err(err) => {
+                eprintln!("Ignoring directive `{}`: {}", s, err);
+                None
+            }
+        }) {
+            env_filter = env_filter.add_directive(directive);
+        }
+    }
+
+    tracing_subscriber::fmt::Subscriber::builder()
+        .with_env_filter(env_filter)
+        .with_writer(std::io::stderr)
+        .init();
+}
+
 pub fn init_test_logger() {
-    let _ = env_logger::Builder::new()
-        .filter_module("tokio_reactor", LevelFilter::Info)
-        .filter_module("tokio_core", LevelFilter::Info)
-        .filter_module("hyper", LevelFilter::Info)
-        .filter(None, LevelFilter::Debug)
-        .try_init();
+    let env_filter = EnvFilter::new("tokio_reactor=info,tokio_core=info,hyper=info,debug");
+    setup_subscriber_from_filter(env_filter);
     init_stop_on_panic();
 }
 
 pub fn init_test_logger_allow_panic() {
-    let _ = env_logger::Builder::new()
-        .filter_module("tokio_reactor", LevelFilter::Info)
-        .filter_module("tokio_core", LevelFilter::Info)
-        .filter_module("hyper", LevelFilter::Info)
-        .filter(None, LevelFilter::Debug)
-        .try_init();
+    let env_filter = EnvFilter::new("tokio_reactor=info,tokio_core=info,hyper=info,debug");
+    setup_subscriber_from_filter(env_filter);
 }
 
 pub fn init_test_module_logger(module: &str) {
-    let _ = env_logger::Builder::new()
-        .filter_module("tokio_reactor", LevelFilter::Info)
-        .filter_module("tokio_core", LevelFilter::Info)
-        .filter_module("hyper", LevelFilter::Info)
-        .filter_module("cranelift_wasm", LevelFilter::Warn)
-        .filter_module(module, LevelFilter::Info)
-        .filter(None, LevelFilter::Info)
-        .try_init();
+    let env_filter =
+        EnvFilter::new("tokio_reactor=info,tokio_core=info,hyper=info,cranelift_wasm=warn,info")
+            .add_directive(format!("{}=info", module).parse().unwrap());
```

Comment:
also this was done because I didn't know how to use RUST_LOG. So may be we don't need this `verbose=network`  mode if we can just use `RUST_LOG=network=info,etc` (just may be need to pipe it through nearup into docker).

Reply:
To pass environment variables to docker we can use `-e RUSTLOG=$RUST_LOG`